src/libDirectoryService/DSBlockPostProcessing.cpp

```diff
@@ -233,38 +229,48 @@ void DirectoryService::UpdateMyDSModeAndConsensusId() {
                                             .GetHashForRandom()
                                             .asBytes());
   }
-  // Check if I am the oldest backup DS (I will no longer be part of the DS
-  // committee)
-  if ((uint32_t)(m_consensusMyID + numOfIncomingDs) >=
-      m_mediator.m_DSCommittee->size()) {
+
+  // Find my new consensus ID.
+  DequeOfNode::iterator it;
```

Comment:
Put the definition into the for loop, can use auto it = m_mediator.m_DSCommittee->begin(), and use the m_mutexDSCommittee to lock the m_DSCommittee before use it.

Reply:
Thanks, relevant iterators are now using auto instead of explicitly declaring iterator type in this commit: https://github.com/Zilliqa/Zilliqa/pull/1587/commits/83971d8f35d9b5f2eb51d743d8cf3957a46c8e62